bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogger.java

```diff
@@ -264,6 +274,8 @@ public EntryLogger(ServerConfiguration conf,
         // but the protocol varies so an exact value is difficult to determine
         this.maxSaneEntrySize = conf.getNettyMaxFrameSizeBytes() - 500;
         this.ledgerDirsManager = ledgerDirsManager;
+        this.conf = conf;
+        entryLogPerLedgerEnabled = conf.isEntryLogPerLedgerEnabled();
```

Comment:
The entryLogManager should be constructed outside of the EntryLogger and passed in as an argument. The entryLogger shouldn't need to know which implementation of the entryLogManager it is using, which is the whole point of having an interface. Also injecting the implementation like this, will make for easier testing, and SortedLedgerStorage would not need to be touched at all (i.e. SortedLedgerStorage can be changed to always create the SingleLog variant).

Reply:
hmm..I considered moving out EntryLogManager, EntryLogManagerForSingleEntryLog and EntryLogManagerForEntryLogPerLedger out of EntryLogger, but  I realized that there is circular dependency. EntryLogger has a EntryLogManager reference, and for certain functionalities of EntryLogManager, it need to have reference of EntryLogger to call its methods in EntryLogManager. So I don’t see much benefits other than just creating new files and new classes (which I'm ok with).